src/DeFuncto.Core/Extensions/Collections.cs

```diff
@@ -33,7 +33,22 @@ public static class Collections
         [Pure]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Option<T> FirstOrNone<T>(this List<T> self) =>
-            self.Select(t => new Box<T>(t)).FirstOrDefault().Apply(Optional).Map(box => box.Value);
+            self. Select(t => new Box<T>(t)).FirstOrDefault().Apply(Optional).Map(box => box.Value);
+
+        [Pure]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Option<T> FirstOrNone<T>(this IQueryable<T> query, Func<T, bool> filter) =>
```

Comment:
Not on IQueryable, it's going to be a nightmare of colissions when we do the EF Core library.